### PR TITLE
Revert "Merge pull request #4025 from greenify/is_strictly_sorted"

### PR DESCRIPTION
**Pull request by DmitryOlshansky**
This reverts commit 1f008535131910849948aa140afd667b90f00227, reversing
changes made to 580d72f3b30cee31996d6b170c39b987a02aabc9.

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -132,89 +130,55 @@ unittest
...
+                !binaryFun!less(r[i], r[i + 1]),
+                "Predicate for isSorted is not antisymmetric. Both" ~
+                        " pred(a, b) and pred(b, a) are true for certain values.");
+            return false;
         }
     }
     else
     {
-        auto ahead = r.save;
+        auto ahead = r;
```
Comment (skoppe) [inline std/algorithm/sorting.d]: I thought the change to `auto ahead = r.save` was a good one.

https://github.com/D-Programming-Language/phobos/pull/4025/files#diff-d25a56391e757d0c5c85b6494c2b2762R163

Comment (DmitryOlshansky) [inline std/algorithm/sorting.d]: This is intended as clean revert, bugfix needs to be its own pull.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by andralex** [comment]
Much appreciated you guys!

**Pull request merged**